### PR TITLE
Add code for Getting started with Firebase on Apple platforms

### PR DESCRIPTION
**Pull request by peterfriese**

### CODE REVIEW

File: fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouriteNumberView.swift [hunk 1]
```diff
@@ -0,0 +1,77 @@
+//
+//  FavouriteNumberView.swift
+//  Shared
+//
+//  Created by Peter Friese on 05.07.22.
+//  Copyright © 2021 Google LLC. All rights reserved.
```
Comment (morganchen12) [inline fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouriteNumberView.swift:6]: nit: I think `All rights reserved` is directly at odds with the license below, but I'm not a lawyer. Also copyright date should be 2022.

File: fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouriteNumberView.swift [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import Combine
+import FirebaseAnalytics
+import FirebaseAnalyticsSwift
+
+class FavouriteNumberViewModel: ObservableObject {
+  @Published var favouriteNumber: Int = 42
```
Comment (morganchen12) [inline fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouriteNumberView.swift:26]: nit: The type can be implied here (number literals default to int if they don't have a decimal component iirc)

File: fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouritesApp.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import SwiftUI
+import Firebase
+import FirebaseCore
+
+class AppDelegate: NSObject, UIApplicationDelegate {
+  func application(_ application: UIApplication,
+                   didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
+    FirebaseApp.configure()
+    Firebase.Analytics.setAnalyticsCollectionEnabled(true)
```
Comment (morganchen12) [inline fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouritesApp.swift:28]: nit: specifying the module here should be unnecessary, and also the top-level Firebase module is deprecated. If you need to specify the module, it should be `FirebaseAnalytics.Analytics`.

File: fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouritesApp.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+import Firebase
+import FirebaseCore
+
+class AppDelegate: NSObject, UIApplicationDelegate {
+  func application(_ application: UIApplication,
+                   didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
```
Comment (morganchen12) [inline fundamentals/apple/gettingstarted/final/Favourites/Shared/FavouritesApp.swift:26]: Specifying `= nil` in the params list creates an overload that's unused.

File: fundamentals/apple/gettingstarted/starter/Favourites/Shared/FavouriteNumberView.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
+//
```
Comment (morganchen12) [inline fundamentals/apple/gettingstarted/starter/Favourites/Shared/FavouriteNumberView.swift:1]: nit: For the starter project, consider leaving placeholder comments where users should be writing code.

**Review by morganchen12 — APPROVED** [review]
Ah, finally getting to write some Combine code :)

File: fundamentals/apple/gettingstarted/starter/Favourites/Shared/FavouriteNumberView.swift (hunk 5 above)
Comment (peterfriese) [reply fundamentals/apple/gettingstarted/starter/Favourites/Shared/FavouriteNumberView.swift:1]: It's literally what code completion gives me:

CleanShot 2022-10-28 at 00 33 25@2x

Also, the code in the function will indeed be called (verified using a breakpoint).

**Pull request merged**